python/modules/IcePy/Types.h

```diff
@@ -209,20 +204,20 @@ class PrimitiveInfo : public TypeInfo
 
     const Kind kind;
 };
-typedef IceUtil::Handle<PrimitiveInfo> PrimitiveInfoPtr;
+using PrimitiveInfoPtr = std::shared_ptr<PrimitiveInfo>;
 
 //
 // Enum information.
 //
-typedef std::map<Ice::Int, PyObjectHandle> EnumeratorMap;
+using EnumeratorMap = std::map<Ice::Int, PyObjectHandle>;
 
-class EnumInfo : public TypeInfo
+class EnumInfo final : public TypeInfo
 {
 public:
 
     EnumInfo(const std::string&, PyObject*, PyObject*);
 
-    virtual std::string getId() const;
+    std::string getId() const final;
 
     virtual bool validate(PyObject*);
```

Comment:
I don't think this needs to use virtual either.

Reply:
I keep changes to a minimum with this PR to avoid conflicts with Benoit updates, once both are merged we need to do another pass with more updates.